server/src/main/java/org/elasticsearch/index/mapper/JsonFieldMapper.java

```diff
@@ -0,0 +1,276 @@
+            throw new UnsupportedOperationException("[fields] is not supported for [" + CONTENT_TYPE + "] fields.");
+        }
+
+        public Builder copyTo(CopyTo copyTo) {
+            throw new UnsupportedOperationException("[copy_to] is not currently supported for ["
+                + CONTENT_TYPE + "] fields.");
+        }
+
+        @Override
+        public JsonFieldMapper build(BuilderContext context) {
+            setupFieldType(context);
+            return new JsonFieldMapper(name, fieldType, defaultFieldType,
+                ignoreAbove, context.indexSettings());
+        }
+    }
+
+    public static class TypeParser implements Mapper.TypeParser {
+        @Override
+        public Mapper.Builder<?,?> parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
+            JsonFieldMapper.Builder builder = new JsonFieldMapper.Builder(name);
+            parseField(builder, name, node, parserContext);
+            for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+                Map.Entry<String, Object> entry = iterator.next();
+                String propName = entry.getKey();
+                Object propNode = entry.getValue();
+                if (propName.equals("ignore_above")) {
+                    builder.ignoreAbove(XContentMapValues.nodeIntegerValue(propNode, -1));
+                    iterator.remove();
+                } else if (propName.equals("norms")) {
+                    TypeParsers.parseNorms(builder, name, propNode);
+                    iterator.remove();
+                } else if (propName.equals("null_value")) {
+                    if (propNode == null) {
+                        throw new MapperParsingException("Property [null_value] cannot be null.");
+                    }
+                    builder.nullValue(propNode.toString());
+                    iterator.remove();
+                }
+            }
+            return builder;
+        }
+    }
+
+    public static final class JsonFieldType extends StringFieldType {
+
+        public JsonFieldType() {
+            setIndexAnalyzer(Lucene.KEYWORD_ANALYZER);
+            setSearchAnalyzer(Lucene.KEYWORD_ANALYZER);
+        }
+
+        private JsonFieldType(JsonFieldType ref) {
+            super(ref);
+        }
+
+        public JsonFieldType clone() {
+            return new JsonFieldType(this);
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            if (omitNorms()) {
+                return new TermQuery(new Term(FieldNamesFieldMapper.NAME, name()));
+            } else {
+                return new NormsFieldExistsQuery(name());
+            }
+        }
+
+        @Override
+        public Query fuzzyQuery(Object value, Fuzziness fuzziness, int prefixLength, int maxExpansions,
```

Comment:
I'm disallowing these more advanced queries for now, but will look into supporting them in a follow-up (tracked on the meta-issue).

In addition to `fuzzy` and `regexp`, I would also like to disallow `wildcard`. However, `WildcardQueryBuilder` doesn't delegate to a method on `MappedFieldType`. This means we attempt to create wildcard queries on non-text or keyword fields, and usually fail because the value can't be parsed. Do you know if there is a reason for this set-up, or if it's just an oversight that would be good to address?

Reply:
I'm not sure regarding why wildcard is different. I suspect its just because we haven't before needed for change the behaviour of wildcard queries based on the field type. I can't see any reason not to change it so we can control the behaviour here though. If we do make the change it might be best to make it directly on master and then pull the change into this branch to disallow it as the change may become difficult to maintain in the feature branch